addons/sms/wizard/sms_composer.py

```diff
@@ -320,10 +321,11 @@ def _prepare_mass_sms_values(self, records):
                 'state': state,
                 'failure_type': failure_type,
             }
-        return result
+        return (result if not self.composition_mode == 'mass' or mass_include_canceled
+                else {res_id: values for res_id, values in result.items() if values.get('state') != 'canceled'})
```

Comment:
Same as in the other commit, might be better to use a regular if-else block here